spec.html

```diff
@@ -436,626 +428,49 @@ <h1>%PrivateNameSet% ( _name_, _object_, _value_ )</h1>
+          1. Let _pn_ be ? ThisPrivateName().
+          1. Let _desc_ be _sym_'s [[Description]] value.
+          1. If _desc_ is *undefined*, return _desc_ be the empty string.
+          1. Otherwise, return _desc_.
         </emu-alg>
-
-        <emu-clause id="sec-private-namedescriptivestring" aoid="PrivateNameDescriptiveString">
-          <h1>Runtime Semantics: PrivateNameDescriptiveString ( _sym_ )</h1>
-          <p>When the abstract operation PrivateNameDescriptiveString is called with argument _sym_, the following steps are taken:</p>
-          <emu-alg>
-            1. Assert: Type(_sym_) is PrivateName.
-            1. Let _desc_ be _sym_'s [[Description]] value.
-            1. If _desc_ is *undefined*, let _desc_ be the empty string.
-            1. Assert: Type(_desc_) is String.
-            1. Return the result of concatenating the strings `"#"` and _desc_.
-          </emu-alg>
-        </emu-clause>
       </emu-clause>
 
-      <emu-clause id="sec-private-name.prototype.valueof">
-        <h1>PrivateName.prototype.valueOf ( )</h1>
+      <emu-clause id="sec-private-name.prototype.tostring">
+        <h1>%PrivateName%.prototype.toString ( )</h1>
         <p>The following steps are taken:</p>
         <emu-alg>
-          1. Return ? ThisPrivateName().
+          1. Throw a *TypeError* exception.
         </emu-alg>
+        <emu-note>
+          Because conversion to a string throws, ToPropertyKey applied to a %PrivateName% object throws as well. This property is important to ensure that Private Names are not incorrectly used by decorators using property access, rather than with their `get` and `set` methods.
+        </emu-note>
       </emu-clause>
 
-      <emu-clause id="sec-private-name.prototype-@@toprimitive">
-        <h1>PrivateName.prototype [ @@toPrimitive ] ( _hint_ )</h1>
-        <p>This function is called by ECMAScript language operators to convert a PrivateName object to a primitive value. The allowed values for _hint_ are `"default"`, `"number"`, and `"string"`.</p>
-        <p>When the `@@toPrimitive` method is called with argument _hint_, the following steps are taken:</p>
-        <emu-alg>
-          1. Return ? ThisPrivateName().
-        </emu-alg>
-        <p>The value of the `name` property of this function is `"[PrivateName.toPrimitive]"`.</p>
+      <emu-clause id="sec-private-name.prototype-@@tostringtag">
+        <h1>PrivateName.prototype [ @@toStringTag ]</h1>
+        <p>The initial value of the @@toStringTag property is the String value `"PrivateName"`.</p>
         <p>This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</p>
       </emu-clause>
 
       <emu-clause id="sec-private-name-this-private-name" aoid=ThisPrivateName>
         <h1>ThisPrivateName()</h1>
         <emu-alg>
-          1. Let _p_ be the *this* value.
-          1. Return ToPrivateName(_p_).
+          1. Let _O_ be the *this* value.
```

Comment:
Can abstract operations access the receiver in this manner? I feel like what I’ve typically seen is that API methods extract the receiver and pass it to abstract ops as an argument.